### PR TITLE
Add missing types on Firestore.Query interface

### PR DESCRIPTION
**Pull request by originerd**
- startAfter
- startAt

### Summary

There are no types for `startAfter` and `startAt` on `Firestore.Query` interface on neither `TypeScript` nor `Flow`. So I added these to `Firestore.Query` interface.

### Checklist

- [x] Supports `Android`
- [x] Supports `iOS`
- [ ] `e2e` tests added or updated in [/tests/e2e/\*](/tests/e2e)
- [ ] Updated the documentation in the [docs repo](https://github.com/invertase/react-native-firebase-docs)
  - **LINK TO DOCS PR HERE**
- [x] Flow types updated
- [x] Typescript types updated

### Test Plan

You could check it via `TypeScript` and`Flow`.

### Release Plan

[CATEGORY][type] [LOCATION] - Add `startAfter` and `startAt` methods to `Firestore.Query` interface on `TypeScript` and `Flow`

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

File: packages/firestore/lib/index.d.ts [hunk 1]
```diff
@@ -1045,6 +1045,88 @@ export namespace Firestore {
...
+    startAfter(snapshot: DocumentSnapshot): Query;
+
+    /**
+     * Creates and returns a new Query that starts after the provided fields relative to the order of
+     * the query. The order of the field values must match the order of the order by clauses of the query.
+     *
+     * #### Example
+     *
+     * ```js
+     * // Get all users who's age is 29 or less
```
Comment (Ehesp) [inline packages/firestore/lib/index.d.ts]: `Get all users who's age is above 30`

File: packages/firestore/lib/index.d.ts [hunk 2]
```diff
@@ -1045,6 +1045,88 @@ export namespace Firestore {
...
+     * #### Example
+     *
+     * ```js
+     * const user = await firebase.firestore().doc('users/alovelace').get();
+     *
+     * // Get all users up to a specific user in order of age
+     * const querySnapshot = await firebase.firestore()
+     *   .collection('users')
+     *   .orderBy('age')
+     *   .startAt(tuser);
```
Comment (Ehesp) [inline packages/firestore/lib/index.d.ts]: `.startAt(tuser);` => `.startAt(user);`

File: packages/firestore/lib/index.d.ts [hunk 3]
```diff
@@ -1045,6 +1045,88 @@ export namespace Firestore {
...
+    startAt(snapshot: DocumentSnapshot): Query;
+
+    /**
+     * Creates and returns a new Query that starts at the provided fields relative to the order of the query.
+     * The order of the field values must match the order of the order by clauses of the query.
+     *
+     * #### Example
+     *
+     * ```js
+     * // Get all users who's age is 30 or less
```
Comment (Ehesp) [inline packages/firestore/lib/index.d.ts]: `Get all users who's age is 30 or above`

**Comment by Ehesp** [comment]
Thanks for this - just a few comments (please replicate onto Flow) :D

**Comment by originerd** [comment]
@Ehesp Thanks for the good points. I applied these.

**Pull request merged**